packages/cli/src/deprecation/deprecation.service.ts

```diff
@@ -89,6 +97,16 @@ export class DeprecationService {
 	}
 
 	mustWarn(envVar: string) {
-		return this.state[envVar]?.mustWarn ?? false;
+		for (const [deprecation, metadata] of this.state.entries()) {
```

Comment:
using a Map and still iterating over all the entries doesn't look great. maybe we can change the Map back to use the `envVar` as the key?

Reply:
Ah, this is for the `mustWarn` function. 

To be honest this function is only used in tests. I was thinking about removing it and asserting the `logger.warn` function in the tests. What do you think?

Reply:
removed it